### PR TITLE
samples/nrf52/power_mgr: rework for clarity and functionality

### PR DESCRIPTION
**Pull request by pabigot**
### samples: nrf52: power_mgr: clean up configuration files

This example retained distinct default and tickless configurations,
although tickless became the default several releases back.  Remove
the tickless configuration, and remove the non-default settings for
BT, DEVICE_POWER_MANAGEMENT, and TIMESLICING, neither of which have
any relevance to this example.

### samples: nrf52: power_mgr: do not skip sleep state 1

Nordic SOC system power only supports deep sleep 1, not sleep state 1.
The sample enabled sleep states 2 and 3, while defining the residency
durations for sleep states 1 and 2.  The residency duration is ignored
if the corresponding sleep state isn't selected.  Reduce confusion by
not skipping sleep states.

This is a rote transformation from state 2 to 1 and state 3 to 2 with
no functional change intended.

### samples: nrf52: power_mgr: rework for clarity and functionality

The original code used a complex sequence that cycled through several
configurations with delays of either 10 to 30 s, requiring signficant
patience to see the effect.  Replace this with iteration through a
sequence of busy-wait/short sleep/medium sleep/long sleep and display
the effect of enabling/disabling sleep states on the residency-based
selection of power state.

Also implement the GPIO-triggered wakeup from deep sleep.

### CODE REVIEW

File: samples/boards/nrf52/power_mgr/Kconfig [hunk 1]
```diff
@@ -3,8 +3,8 @@
 config NRF5_POWER_MGMT_EXAMPLE
 	bool
 	default y
+	select HAS_SYS_POWER_STATE_SLEEP_1
 	select HAS_SYS_POWER_STATE_SLEEP_2
```
Comment (mnkp) [inline samples/boards/nrf52/power_mgr/Kconfig]: As noted in the PR description the NRF52 doesn't support 'Sleep' states, only 'Deep Sleep 1' state. We shouldn't enable states that are not supported. This is going to be confusing for anyone trying to understand the design of Zephyr power management subsystem. The `HAS_SYS_POWER_STATE_` options should be set by the respective SoC not the sample app. Sample applications are there to provide guidance, showcase usage. This example is incorrect.

When run with `CONFIG_SOC_LOG_LEVEL_DBG=y` the app generates the output
```
3 power states are supported
[  0.036]: Sys states for 03: +S1 +S2
[  0.039]: Busy-wait 1 s
[  1.036]: Sleep 1499 ms
[  2.537]: Sleep 2500 ms
[  2.540]:  --> Entering to SYS_POWER_STATE_SLEEP_1 state.
[  5.040]:  --> Exited from SYS_POWER_STATE_SLEEP_1 state.
[  5.045]: Sleep 3500 ms
[00:00:05.045,440] <dbg> soc._sys_pm_power_state_exit_post_ops: Unsupported power state 0
[00:00:05.047,760] <dbg> soc.sys_set_power_state: Unsupported power state 0
[  6.061]:  --> Entering to SYS_POWER_STATE_SLEEP_1 state.
[  8.547]:  --> Exited from SYS_POWER_STATE_SLEEP_1 state.
[  8.552]: Sys states for 02: -S1 +S2
[  8.556]: Busy-wait 1 s
[  9.552]: Sleep 1499 ms
[00:00:08.552,917] <dbg> soc._sys_pm_power_state_exit_post_ops: Unsupported power state 0
[00:00:09.563,507] <dbg> soc.sys_set_power_state: Unsupported power state 0
[ 11.054]: Sleep 2500 ms
[ 13.556]: Sleep 3500 ms
[ 13.558]:  --> Entering to SYS_POWER_STATE_SLEEP_2 state.
[ 17.058]:  --> Exited from SYS_POWER_STATE_SLEEP_2 state.
[ 17.063]: Sys states for 01: +S1 -S2
[ 17.067]: Busy-wait 1 s
[ 18.063]: Sleep 1499 ms
[00:00:17.063,781] <dbg> soc._sys_pm_power_state_exit_post_ops: Unsupported power state 1
[00:00:18.074,157] <dbg> soc.sys_set_power_state: Unsupported power state 1
```

**Review by mnkp — CHANGES_REQUESTED** [review]

File: samples/boards/nrf52/power_mgr/README.rst [hunk 2]
```diff
@@ -6,29 +6,23 @@ nRF5x Power management demo
...
-power management infrastructure to enter into Low Power state.
+This sample demonstrates a power manager app that uses the Zephyr system
+power management infrastructure to manage multiple low-power states that
+are conditional on the duration of sleep.  The power mode selection is
+done automatically by residency-based policy implemented by the Zephyr
+Power Management Subsystem.
 
-This app will cycle through the following power schemes each time the system
-enters idle state:
+After all combinations have been completed the system falls of the end
```
Comment (dbkinder) [inline samples/boards/nrf52/power_mgr/README.rst]: After all combinations **of what** have been completed?

``the system falls off the end of main()`` is an oddly non-technical description, maybe instead:
``the main() function returns``  (also there's a misspelling of ``off``)

What does pressing button 1 do after the main() function returns?  Does it start the main() function again?  If you press button 1 while the sample is running, what happens?

**Review by dbkinder — CHANGES_REQUESTED** [review]
seems to be missing important details about what the sample does.

**Comment by pabigot** [comment]
Never mind.

**Pull request closed without merging**